packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -262,67 +286,82 @@ const Dropdown = ({
 
   const closeIcons = (
     <ValueIconContainer>
-      {clearable && (
+      {onClear && values.length > 0 && (
         <CloseIconContainer
           onClick={handleClear}
           onFocus={(e: React.FocusEvent) => e.stopPropagation()}
           tabIndex={tabIndex}
         >
-          <Icon path={mdiClose} size={0.75} />
+          <Icon path={mdiClose} size="1em" />
         </CloseIconContainer>
       )}
-      <Icon path={state.isOpen ? mdiMenuUp : mdiMenuDown} size={0.75} />
+      <Icon path={isOpen ? mdiMenuUp : mdiMenuDown} size="1.25em" />
     </ValueIconContainer>
   );
 
   return (
     <StyledContainer
-      data-testid={`${state.id}-valueContainer`}
+      data-testid={`${name}-valueContainer`}
       elevation={elevation}
-      isOpen={state.isOpen}
-      id={`${state.id}-valueContainer`}
+      isOpen={isOpen}
+      id={`${name}-valueContainer`}
       name={name}
       onBlur={handleBlur}
       onFocus={(e: React.FocusEvent) => {
         e.preventDefault();
-        setState(curState => ({ ...curState, isOpen: true }));
+        setIsOpen(true);
       }}
       tabIndex={tabIndex}
     >
       <Button
         StyledContainer={StyledValueContainer}
         containerProps={{
-          modalIsOpen: state.isOpen,
+          modalIsOpen: isOpen,
         }}
         color={color}
         onClick={(e: React.MouseEvent) => e.preventDefault()}
-        variant={type}
+        variant={variant}
       >
         <StyledValueItem>
-          {((values.length && values) || state.selectedValues).join(', ')}
+          {values
+            .filter(val => val !== undefined)
+            .map((val, i) =>
+              optionsHash[val] !== undefined ? (
+                <>
+                  {i !== 0 && ', '}
+                  {optionsHash[val].optionValue}
+                </>
+              ) : (
```

Comment:
I felt like .join reads a bit cleaner, was there a problem with .join?

Reply:
You can't .join with array of ReactNodes apparently. Or at least it failed when I tried it.

Reply:
Ahh I didn't pick up that optionValue was no longer just a string. Makes sense.